src/site/stages/build/index.js

```diff
@@ -38,6 +39,40 @@ const addSubheadingsIds = require('./plugins/add-id-to-subheadings');
 
 function defaultBuild(BUILD_OPTIONS) {
   const smith = Metalsmith(__dirname); // eslint-disable-line new-cap
+
+  // Override the normal use function to provide timing and logging information
+  smith._use = smith.use;
+  let stepCount = 0;
+  smith.use = function use(plugin, description) {
+    const step = ++stepCount;
+    if (!description) return smith._use(plugin);
+
+    let timerStart;
+
+    /* eslint-disable no-console */
+    return smith
+      ._use(() => {
+        console.log(chalk.cyan(`\nStep ${step} start: ${description}`));
+        timerStart = process.hrtime.bigint();
+      })
+      ._use(plugin)
+      ._use(() => {
+        const time = (process.hrtime.bigint() - timerStart) / 1000000n;
```

Comment:
Learned about [BigInts](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt). Turns out you can't do arithmetic with mixed `BigInts` and `number`s, hence the `n` in `1000000n`.